### ISSUE TITLE
expose createSlice

### ISSUE THREAD

**Issue by wilzbach** [9il/dip80-ndslice#17]
I found the following pattern hidden in an unittest - why is this not public?

This would be very useful for many array creation, because it would solve NumPy's popular `zeros`, `ones` and with `createSlice(3,3).diag[] = 1` also `identitiy`

Moreover I would suggest to use a default type or add some convenience wrappes like the one mentioned from [NumPy](https://docs.scipy.org/doc/numpy-1.10.1/reference/routines.array-creation.html)

```
auto createSlice(T, Lengths...)(Lengths lengths)
{
    return createSlice2!(T, Lengths.length)(cast(size_t[Lengths.length])[lengths]);
}

///ditto
auto createSlice2(T, size_t N)(auto ref size_t[N] lengths)
{
    size_t length = lengths[0];
    foreach (len; lengths[1 .. N])
        length *= len;
    return new T[length].sliced(lengths);
}


pure nothrow unittest
{
    auto slice = createSlice!int(5, 6, 7);
    assert(slice.length == 5);
    assert(slice.elementsCount == 5 * 6 * 7);
    static assert(is(typeof(slice) == Slice!(3, int*)));
}
```

For what it is worth I also saw a `makeSlice` which is based on the new allocator.

```
import std.experimental.allocator;

auto makeSlice(T, Allocator, Lengths...)(auto ref Allocator alloc, Lengths lengths)
{
    enum N = Lengths.length;
    struct Result { T[] array; Slice!(N, T*) slice; }
    size_t length = lengths[0];
    foreach (len; lengths[1 .. N])
        length *= len;
    T[] a = alloc.makeArray!T(length);
    return Result(a, a.sliced(lengths));
}

unittest
{
    auto tup = makeSlice!int(theAllocator, 2, 3, 4);

    static assert(is(typeof(tup.array) == int[]));
    static assert(is(typeof(tup.slice) == Slice!(3, int*)));

    assert(tup.array.length           == 24);
    assert(tup.slice.elementsCount    == 24);
    assert(tup.array.ptr == &tup.slice[0, 0, 0]);

    theAllocator.dispose(tup.array);
}
```

**Comment by 9il** [comment]
Hi @greenify 

I plan to add a module for allocation functions. So this function will be implemented

**Comment by 9il** [comment]
fixed with #38
